ActiveLabeler-main/ActiveLabeler.py

```diff
@@ -0,0 +1,181 @@
+
+        query -- A num_images X num_classes sized numpy array. This numpy array has the softmax probability
+                    of each image. These softmax probs are predicted by the latest finetune model. Type: numpy array
+
+        N -- Number of images that should be in the subset. Type: integer
+
+        strategy_params -- A dictionary of strategy specific args. Type: Dictionary
+
+        Returns list/numpy array of the index of images in the subset.
+        """
+        if name == "gaussian":
+
+            def gaussian(x, mean, sigma):
+                a = 1 / (sigma * np.sqrt(2 * np.pi))
+                exp_val = -(1 / 2) * (((x - mean) ** 2) / (sigma ** 2))
+                tmp = list(a * np.exp(exp_val))[0]
+                return tmp
+
+            def get_gaussian_weights(confidences):
+                mean = 0.5
+                sigma = 0.2
+                return [gaussian(conf_i, mean, sigma) for conf_i in confidences]
+
+            def get_samples(image_list, confidences, number_of_samples):
+                sample_wts = get_gaussian_weights(confidences)
+                sample_wts_norm = np.array(sample_wts)
+                sample_wts_norm /= sample_wts_norm.sum()
+                picked_samples = np.random.choice(
+                    image_list, replace=False, size=number_of_samples, p=sample_wts_norm
+                )
+                return picked_samples
+
+            def get_index(samples, dataset):
+                data_index = []
+                for i in samples:
+                    data_index.append(dataset.index(i))
+                data_index = np.array(data_index)
+                return dataset[data_index]
+
+            selected_samples = get_samples(self.images_path, query, N)
+            # selected_embeddinds = get_index(selected_samples, self.embeddings)
+            return selected_samples
+        elif name == "uncertainty":
+            tmp_query = np.array([query[i][0] for i in range(len(query))])
+            difference_array = np.absolute(tmp_query - 0.5)
+            return difference_array.argsort()[:N]
+        elif name == "uncertainty_balanced":
+            tmp_query = np.array([query[i][0] for i in range(len(query))])
+            difference_array = tmp_query - 0.5
+            sorted_diff = difference_array.argsort()
+            idx_0 = np.absolute(difference_array).argsort()[0]
+            idx_sorted_0 = list(sorted_diff).index(idx_0)
+
+            #todo logs
+            print("tmp_query- probs - sorted", sorted(tmp_query))
+            print("diff arr ", difference_array)
+            print("diff arr sorted ", np.sort(difference_array))
+            print("middle value idx", idx_sorted_0)
+            print("middle value ", difference_array[idx_sorted_0])
+
+            N = N - 1
+            # take n/2 from either side of index_sorted_0, less ele
+            print("splitting idices", (idx_sorted_0 - (N // 2) - 1), idx_sorted_0)
```

Comment:
Is this something a user of this library would want to see in their logs? If so, please fix the spelling of the message and perhaps add more clarity as to what is being output.